src/imputors/locf.jl

```diff
@@ -12,30 +12,59 @@ See also:
 existing observation to carry forward. As a result, this method does not guarantee
 that all missing values will be imputed.
 
+# Keyword Arguments
+The following additional keyword arguments can be applied during impute:
+* `limit=Inf`: Limits the amount of consecutive missing values to replace.
+* `index_values::AbstractVector`: A sorted vector of index information for the data.
+  This is compared when using `limit` to limit replacement of data at irregular intervals
+  or by other types (eg: timestamps).
+
 # Example
 ```jldoctest
 julia> using Impute: LOCF, impute
 
-julia> M = [1.0 2.0 missing missing 5.0; 1.1 2.2 3.3 missing 5.5]
-2×5 Matrix{Union{Missing, Float64}}:
- 1.0  2.0   missing  missing  5.0
- 1.1  2.2  3.3       missing  5.5
+julia> M = [1.0 2.0 missing missing missing 6.0; 1.1 missing missing 4.4 5.5 6.6]
+2×6 Matrix{Union{Missing, Float64}}:
+ 1.0  2.0       missing   missing   missing  6.0
+ 1.1   missing  missing  4.4       5.5       6.6
 
 julia> impute(M, LOCF(); dims=:rows)
-2×5 Matrix{Union{Missing, Float64}}:
- 1.0  2.0  2.0  2.0  5.0
- 1.1  2.2  3.3  3.3  5.5
+2×6 Matrix{Union{Missing, Float64}}:
+ 1.0  2.0  2.0  2.0  2.0  6.0
+ 1.1  1.1  1.1  4.4  5.5  6.6
+
+julia> impute(M,  LOCF(); dims=:rows, limit=2)
+2×6 Matrix{Union{Missing, Float64}}:
+ 1.0  2.0  2.0  2.0   missing  6.0
+ 1.1  1.1  1.1  4.4  5.5       6.6
+
+julia> impute(M,  LOCF(); dims=:rows, limit=1.5, index_values=[1.0, 1.5, 2.0, 3.0, 3.5, 4.0])
+2×6 Matrix{Union{Missing, Float64}}:
+ 1.0  2.0  2.0  2.0   missing  6.0
+ 1.1  1.1  1.1  4.4  5.5       6.6
 ```
 """
 struct LOCF <: Imputor end
 
-function _impute!(data::AbstractVector{Union{T, Missing}}, imp::LOCF) where T
+function _impute!(
+    data::AbstractVector{Union{T, Missing}},
+    imp::LOCF;
+    limit=Inf,
```

Comment:
Hmm, I wonder if this should actually be a field in the `Imputor` type? It seems like a setting that isn't really coupled to the data? Also, we probably want it to be a field of type `Union{Nothing, UInt}` since it's really just a running `count`?

Reply:
> we probably want it to be a field of type Union{Nothing, UInt}

That would defeat our purpose of using it to limit a collection of (possibly irregularly spaced) time series data by seconds.

Reply:
Hmm, maybe this could be generalized into `LOCF` and `WeightedLOCF`? Similar to the `WeightedSubstitute`? https://github.com/invenia/Impute.jl/blob/master/src/imputors/substitute.jl#L84

Reply:
The `index_values` aren't really measures of importance as much as a representation of distance between values. I'm not sure weight is the right name for them.

Reply:
The use case for this change is to be able to do something like `impute(mykeyedarray,  LOCF(); dims=:rows, limit=Hour(1), index_values=mykeyedarray.timestamp)`

Reply:
Kinda feels like we're solving two different problem here. I feel like LOCF implies linear spacing and perhaps you may have some importance weighting schema of observations. Handling something like "irregularly spaced` timeseries seems like something that should be addressed externally to the LOCF algorithm. For example, inserting our irregularly spaced values into a regularly spaced `KeyedVector` with the index_values and then calling LOCF on the result. I guess there's the risk that you're taking up too much space if the data is very sparse... but that's gonna be an issue for LOCF accuracy anyways.

Reply:
I suppose I'll continue using an internal function then.